modules/couchbase/collect.go

```diff
@@ -0,0 +1,159 @@
+package couchbase
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/netdata/go.d.plugin/agent/module"
+	"github.com/netdata/go.d.plugin/pkg/web"
+)
+
+const (
+	urlPathBucketsStats = "/pools/default/buckets"
+)
+
+func (cb *Couchbase) collect() (map[string]int64, error) {
+
+	ms := cb.scrapeCouchbase()
+	if ms.empty() {
+		return nil, nil
+	}
+	collected := make(map[string]int64)
+	err := cb.collectBasicStats(collected, ms)
+	if err != nil {
+		return nil, fmt.Errorf("error on creating a connection: %v", err)
+	}
```

Comment:
`collectBasicStats` shouldnt return an error (according implementation)

i think this is some leftovers, because error text is

> error on creating a connection: %v